File: src/main/java/client/handlers/BlockHandler.java
package client.handlers;

import client.renderables.BlockRender;
import common.block.Block;
import common.world.World;
public class BlockHandler {

    public BlockHandler() {
    }

    public BlockRender handleBlockRenderer(World world){
        return null;
    }

    public BlockRender handleBlockRenderer(Block block, int x, int y) {
        System.out.println(block.getName());
        return new BlockRender(block, x, y);
    }

}

File: src/main/kotlin/client/handlers/BlockHandler.kt
package client.handlers

import client.renderables.BlockRender
import common.block.Block
import common.world.World

@Deprecated("Won't be needed for the future.")
class BlockHandler {

    fun handleBlockRenderer(world: World): BlockRender? {
        return null
    }

    companion object {

        @JvmStatic
        fun handleBlockRenderer(block: Block, x: Int, y: Int): BlockRender {
            return BlockRender(block, x, y)
        }
    }

}
